### ISSUE TITLE
[Docs] Introduce function parameters in the function page.

### ISSUE THREAD

**Issue by RichardCoppin** [modular/mojo#3787]
### Where is the problem?

https://docs.modular.com/mojo/manual/functions

### What can we do better?

My proposal is to introduce function parameters up front in the initial introduction of the function `fn` syntax.

Not doing so, places the reading with a mental "broken link". Anyone coming across the function parameter syntax in a code base or API reference for the first time will not be able to ascertain what this arcane syntax actually means.

Secondly, the footnote refers to the term `parameters` without any explanation of what that term means.
![image](https://github.com/user-attachments/assets/063300ed-caca-4322-a917-34488284624b)

Once the function parameters have been introduced then place the hyperlink for the remainder of the discussion of parameters versus arguments.

## TL;DR: 
Function parameters need to be introduced here:
![image](https://github.com/user-attachments/assets/2b4593de-9c88-4e4a-9f50-8a5991be37cf)

**Comment by scottamain** [comment]
Sorry for the long delay! This has since been improved with a new section called the [Anatomy of a function](https://docs.modular.com/mojo/manual/functions/#anatomy-of-a-function).
